src/qd8-f16-qb4w-gemm/gen/qd8-f16-qb4w-gemm-1x16-minmax-neonfp16arith-mlal-lane-prfm.c

```diff
@@ -0,0 +1,327 @@
+                const int8x8_t vb01234567c4 = vshl_n_s8(vb01234567c45, 4);
+                const int8x8_t vb01234567c5 = vand_s8(vb01234567c45, vmask);
+                const int8x8_t vb89ABCDEFc4 = vshl_n_s8(vb89ABCDEFc45, 4);
+                const int8x8_t vb89ABCDEFc5 = vand_s8(vb89ABCDEFc45, vmask);
+                const int16x8_t vxb01234567c4 = vmovl_s8(vb01234567c4);
+                const int16x8_t vxb89ABCDEFc4 = vmovl_s8(vb89ABCDEFc4);
+
+                vacc0x0123 = vmlal_lane_s16(vacc0x0123, vget_low_s16(vxb01234567c4), vget_high_s16(vxa0), 0);
+                vacc0x4567 = vmlal_lane_s16(vacc0x4567, vget_high_s16(vxb01234567c4), vget_high_s16(vxa0), 0);
+                vacc0x89AB = vmlal_lane_s16(vacc0x89AB, vget_low_s16(vxb89ABCDEFc4), vget_high_s16(vxa0), 0);
+                vacc0xCDEF = vmlal_lane_s16(vacc0xCDEF, vget_high_s16(vxb89ABCDEFc4), vget_high_s16(vxa0), 0);
+
+                if (k >= 6 * sizeof(int8_t)) {
+                  const int16x8_t vxb01234567c5 = vmovl_s8(vb01234567c5);
+                  const int16x8_t vxb89ABCDEFc5 = vmovl_s8(vb89ABCDEFc5);
+
+                  vacc0x0123 = vmlal_lane_s16(vacc0x0123, vget_low_s16(vxb01234567c5), vget_high_s16(vxa0), 1);
+                  vacc0x4567 = vmlal_lane_s16(vacc0x4567, vget_high_s16(vxb01234567c5), vget_high_s16(vxa0), 1);
+                  vacc0x89AB = vmlal_lane_s16(vacc0x89AB, vget_low_s16(vxb89ABCDEFc5), vget_high_s16(vxa0), 1);
+                  vacc0xCDEF = vmlal_lane_s16(vacc0xCDEF, vget_high_s16(vxb89ABCDEFc5), vget_high_s16(vxa0), 1);
+
+                }
+              }
+            }
+          }
+        }
+      }
+
+    const float32x4_t vfilter_output_scale0123 = vld1q_f32(w); w = (const float*) w + 4;
+    const float32x4_t vfilter_output_scale4567 = vld1q_f32(w); w = (const float*) w + 4;
+    const float32x4_t vfilter_output_scale89AB = vld1q_f32(w); w = (const float*) w + 4;
+    const float32x4_t vfilter_output_scaleCDEF = vld1q_f32(w); w = (const float*) w + 4;
+
+    const float32x4_t vf0x0123 = vcvtq_f32_s32(vacc0x0123);
+    const float32x4_t vf0x4567 = vcvtq_f32_s32(vacc0x4567);
+    const float32x4_t vf0x89AB = vcvtq_f32_s32(vacc0x89AB);
+    const float32x4_t vf0xCDEF = vcvtq_f32_s32(vacc0xCDEF);
+
+    vout0x0123 = vfmaq_f32(vout0x0123, vf0x0123, vfilter_output_scale0123);
+    vout0x4567 = vfmaq_f32(vout0x4567, vf0x4567, vfilter_output_scale4567);
+    vout0x89AB = vfmaq_f32(vout0x89AB, vf0x89AB, vfilter_output_scale89AB);
+    vout0xCDEF = vfmaq_f32(vout0xCDEF, vf0xCDEF, vfilter_output_scaleCDEF);
+    }
+    float32x4_t one_sixteenth = vdupq_n_f32(1/16.0);
+    vout0x0123 = vmulq_f32(vout0x0123, one_sixteenth);
+    vout0x4567 = vmulq_f32(vout0x4567, one_sixteenth);
+    vout0x89AB = vmulq_f32(vout0x89AB, one_sixteenth);
+    vout0xCDEF = vmulq_f32(vout0xCDEF, one_sixteenth);
+
+    const float32x4_t vinput_scale0 = vld1q_dup_f32(&quantization_params[0].inv_scale);
```

Comment:
Can we trade 4x`vmulq_f32` with a single scalar `mul` here??
```suggestion
    const float32x4_t vinput_scale0 = vld1q_dup_f32(&quantization_params[0].inv_scale * (1/16.0));
```

Reply:
I benchmarked it and it turns out it very slightly slower across the board on M1 Pro. I'm a little suprised since there can be a lot of muls. I can provide more info (numbers and code) if you're interested.

Reply:
Ok.. The other option is to move this to packing - see [this](https://github.com/google/XNNPACK/pull/6576#discussion_r1645426749) for more details.

Reply:
That's a good idea. Should be universally faster. I'll try it.

Reply:
FYI We discussed this offline and this should not be a blocker to merging this PR. We will handle moving the 1/16 factor as a separate PR.